nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2Client.java

```diff
@@ -135,6 +183,94 @@ public boolean isValidWith(String clientId, String clientSecret) {
+     * @throws NuxeoException if oAuth2Client is not valid
+     * @since 11.1
+     */
+    public static DocumentModel updateDocument(DocumentModel documentModel, OAuth2Client oAuth2Client) {
+        requireNonNull(documentModel, "documentModel model is required");
+        documentModel.setProperties(OAUTH2CLIENT_SCHEMA, OAuth2Client.getDocumentData(oAuth2Client));
+        return documentModel;
+    }
+
+    /**
+     * Get document properties from {@link OAuth2Client}.
+     *
+     * @param oAuth2Client the oAuth2Client
+     * @return the {@code Map} structure of {@code OAuth2Client}
+     * @throws NullPointerException if the oAuth2Client is {@code null}
+     * @throws NuxeoException if oAuth2Client is not valid
+     * @since 11.1
+     */
+    public static Map<String, Object> getDocumentData(OAuth2Client oAuth2Client) {
+        validate(oAuth2Client);
+        Map<String, Object> values = new HashMap<>();
+        values.put(NAME_FIELD, oAuth2Client.getName());
+        values.put(ID_FIELD, oAuth2Client.getId());
+        values.put(REDIRECT_URI_FIELD, StringUtils.join(oAuth2Client.getRedirectURIs(), REDIRECT_URI_SEPARATOR));
+        values.put(AUTO_GRANT_FIELD, oAuth2Client.isAutoGrant());
+        values.put(ENABLED_FIELD, oAuth2Client.isEnabled());
+        if (StringUtils.isNotEmpty(oAuth2Client.getSecret())) {
+            values.put(SECRET_FIELD, oAuth2Client.getSecret());
+        }
+        return values;
+    }
+
+    /**
+     * Validate the {@link OAuth2Client}. An {@code OAuth2Client} is valid if and only if
+     * <ul>
+     * <li>It is not {@code null}</li>
+     * <li>The required fields are fill in: {@link #getId()},{@link #getName()},{@link #getRedirectURIs()}</li>
+     * <li>The {@link #getRedirectURIs()} is a valid URI, {@link #isRedirectURIValid(String)}</li>
+     * </ul>
+     *
+     * @param oAuth2Client the not null oAuth2Client to validate
+     * @throws NullPointerException if the oAuth2Client is {@code null}
+     * @throws NuxeoException if oAuth2Client is not valid
+     * @since 11.1
+     */
+    public static void validate(OAuth2Client oAuth2Client) {
+        requireNonNull(oAuth2Client, "oAuth2Client is required");
+        String message = null;
+        if (StringUtils.isEmpty(oAuth2Client.getName())) {
+            message = "Client name is required";
+        } else if (StringUtils.isEmpty(oAuth2Client.getId())) {
+            message = "Client Id is required";
+        } else if (oAuth2Client.getRedirectURIs().isEmpty()) {
+            message = "Redirect URIs is required";
+        } else if (!oAuth2Client.getRedirectURIs().isEmpty()) {
+            message = oAuth2Client.getRedirectURIs()
+                                  .stream()
+                                  .filter(uri -> !isRedirectURIValid(uri))
+                                  .findAny()
+                                  .map(uri -> String.format("'%s' is not a valid URI", uri))
```

Comment:
"is not a valid redirect URI" (URI could be technically valid but not in terms of an OAuth 2 redirect URI)